### PR TITLE
[SYCL] Support lambda functions passed to reduction

### PR DESCRIPTION
**Pull request by v-klochkov**
Signed-off-by: Vyacheslav N Klochkov <vyacheslav.n.klochkov@intel.com>

### CODE REVIEW

File: sycl/test/reduction/reduction_nd_lambda.cpp [hunk 1]
```diff
@@ -0,0 +1,71 @@
+// UNSUPPORTED: cuda
+// OpenCL C 2.x alike work-group functions not yet supported by CUDA.
```
Comment (alexbatashev) [inline sycl/test/reduction/reduction_nd_lambda.cpp:2]: I'd like to see a bit more clarification here. Are block reductions missing from CUDA itself or CUDA plugin does not expose them?

Comment (v-klochkov) [reply sycl/test/reduction/reduction_nd_lambda.cpp]: Hi Alex, I updated the comment by copying it from here: https://github.com/intel/llvm/blame/sycl/sycl/test/reduction/reduction_nd_s0_dw.cpp#L2 and adding a note about intel::reduce() that is not supported yet by CUDA. 
I don't have any other details regarding CUDA support for that feature.
Thank you!

**Comment by v-klochkov** [comment]
> Aside from the failing test, LGTM.

John, thank you for the quick code-review! I fixed the test.

File: sycl/test/reduction/reduction_nd_lambda.cpp (hunk 1 above)
Comment (Pennycook) [reply sycl/test/reduction/reduction_nd_lambda.cpp]: > I don't have any other details regarding CUDA support for that feature.

It's implementable, but not in the plugin yet.  CUDA doesn't have native support for equivalents to everything in the GroupAlgorithms extension, but it did recently add support for reductions as part of its [cooperative group functionality](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#collectives-cg-reduce).

Comment (v-klochkov) [reply sycl/test/reduction/reduction_nd_lambda.cpp]: Thank you for the link. The updated comment is explanatory enough, I think:
// Reductions use work-group builtins (e.g. intel::reduce()) not yet supported
// by CUDA.

Comment (Pennycook) [reply sycl/test/reduction/reduction_nd_lambda.cpp]: I completely agree.  I was just providing some more background for @alexbatashev.

**Pull request merged**